avocado/utils/partition.py

```diff
@@ -50,30 +50,29 @@ def __str__(self):
                                       super(PartitionError, self).__str__())
 
 
-class MtabLock(object):
+class MtabLock(filelock.FileLock):
```

Comment:
Well, at least for me seems that we should be using composition over inheritance here[1]. On my understanding `MtabLock` uses a `FileLock` and do not *is a* `FileLock`. So it's a composition relation. A smell about that is the changing API of the initialization method. On `MtabLock` the file name argument was dropped.

I do not think it is going to be too different from what you've proposed. I've tried a quick modification from your PR like:

```python
class MtabLock(object):
    def __init__(self, timeout=60):
        self.timeout = timeout
        self.mtab = None
        self.device = 'etc/mtab'
        self.lock_filename = os.path.join(tempfile.gettempdir(), 'some_name')
        self.file_lock = filelock.FileLock(self.lock_filename, timeout=self.timeout)

    def __enter__(self):
        try:
            self.file_lock.__enter__()
        except (filelock.LockFailed, filelock.AlreadyLocked) as e:
            reason = "Unable to obtain '/etc/mtab' lock in %ds" % self.timeout
            raise PartitionError(self, reason, e)
        self.mtab = open(self.device)
        return self

    def __exit__(self, exc_type, exc_value, exc_traceback):
        if self.mtab:
            self.mtab.close()
        self.file_lock.__exit__()
```

[1] - https://en.wikipedia.org/wiki/Composition_over_inheritance

Reply:
IMHO, as the demos wiki provides, composition over inheritance is better in scenario like using inheritance as composition(as the demo wiki provides: to use multiple inheritance to implement class `Player`). As for `MtabLock`, if there is a need to modify the locking scheme someday, composition is better since it forms loose coupling.